### DISCUSSION TITLE
How to deal with performance hit when working with more than 3 fonts

### DISCUSSION THREAD

**Discussion by urdeveloper** [visgl/deck.gl#6545] [Q&A] [answered]
When working with Text Layers which involves more than three unique  combination of font family and font weight, I've noticed performance degradation. 

It turns out Font Atlas Manager has an internal LRU cache for storing generated font atlases and the cache capacity is 3. It means it only caches 3 font atlases at any time. So if you have 5 unique combinations of font family and font weight, Font Atlas Manager keeps evicting the cached atlases to accommodate for the new ones. 

And since creating font atlas is very expensive specially when SDF is used, you'd notice the performance whenever you navigate to an area with features that don't use the cached fonts. 

How would you deal with such problem?

Is making the LRU cache size configurable something you'd be open to? I'll be more than happy to come up with a PR.

**Comment by Pessimistress** [comment, marked as answer]
Fix published in v8.6.7

**Reply by urdeveloper** [reply]
Thanks!
